### ISSUE TITLE
cmd/cgo: undefined reference when using referencing static functions

### ISSUE THREAD

**Issue by abacabadabacaba** [golang/go#34831]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.13.1 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/path/to/home/.cache/go-build"
GOENV="/path/to/home/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/path/to/home/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/lib/go-1.13"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go-1.13/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build344992079=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?

I tried to compile the following program:
```go
package main

// static void f() { }
import "C"

func main() {
	_ = C.f
}
```

### What did you expect to see?

I expected it to compile successfully.

### What did you see instead?

<pre>
# _/path/to/test
/usr/bin/ld: $WORK/b001/_cgo_main.o:/tmp/go-build/cgo-generated-wrappers:2: undefined reference to `f'
collect2: error: ld returned 1 exit status
</pre>

**Comment by ianlancetaylor** [comment]
Note that calling `C.f` works fine.

**Comment by shawn-xdji** [comment]
> Note that calling `C.f` works fine.

cgo generates wrapper of a 'C.f' even it's 'static', is it expected behavior of cgo?

**Comment by ianlancetaylor** [comment]
Yes.  But please take discussions of how cgo works to golang-nuts, not this issue.  Thanks.

**Comment by AdamSLevy** [comment]
How do you work around this?

**Comment by ianlancetaylor** [comment]
@AdamSLevy Use a Go function literal.  For example,

    _ = func() { C.f() }

**Comment by qmuntal** [comment]
Possibly related to #19836

**Comment by bcmills** [comment]
Duplicate of #19836
